python/triqs/gf/gf.py

```diff
@@ -806,7 +806,14 @@ def x_data_view(self, x_window=None, flatten_y=False):
             If flatten_y is True and dim is (1, 1, *) it returns a 1d numpy array.
         """
 
-        X = [x.imag for x in self.mesh] if isinstance(self.mesh, meshes.MeshImFreq) else [x for x in self.mesh]
+        if isinstance(self.mesh, meshes.MeshReFreq):
+            X = np.linspace(self.mesh.w_min, self.mesh.w_max, len(self.mesh))
```

Comment:
I don't think it is guaranteed that this generates the same floating point numbers as `[x.real for x in self.mesh]` ?

Reply:
hm, yes that is probably true. Are you concerned about exact equality of the values? Is it necessary? But okay then we need to loop again right? I hoped to make this method also a bit faster by avoiding for loops.

Reply:
I have moved the generation of the values array to C++, please have a look at the changes.